File: backend/src/models/plant.ts
import mongoose, {Schema} from 'mongoose';

const plantSchema: Schema = new Schema({
  name: {
    type: String,
    required: true,
  },
  otherNames: {
    type: String,
    required: false,
  },
  description: {
    type: String,
    required: false,
  },
  plantSeason: {
    type: String,
    required: false,
  },
  harvestSeason: {
    type: String,
    required: false,
  },
  pruneSeason: {
    type: String,
    required: false,
  },
  tips: {
    type: String,
    required: false,
  },
});

export const Plant = mongoose.model('Plant', plantSchema);

File: plants/src/model/plant.go
package model

// Plant entity
type Plant struct {
	ID            string  `json:"_id" bson:"_id"`
	Name          string  `json:"name" bson:"name"`
	OtherNames    *string `json:"otherNames" bson:"otherNames"`
	Description   *string `json:"description" bson:"description"`
	PlantSeason   *string `json:"plantSeason" bson:"plantSeason"`
	HarvestSeason *string `json:"harvestSeason" bson:"harvestSeason"`
	PruneSeason   *string `json:"pruneSeason" bson:"pruneSeason"`
	Tips          *string `json:"tips" bson:"tips"`
}
